alacritty/src/config/bindings.rs

```diff
@@ -1010,19 +1010,20 @@ impl<'a> Deserialize<'a> for RawBinding {
                                 return Err(<V::Error as Error>::duplicate_field("key"));
                             }
 
-                            let val = map.next_value::<SerdeValue>()?;
-                            if val.is_u64() {
-                                let scancode = val.as_u64().unwrap();
-                                if scancode > u64::from(u32::MAX) {
-                                    return Err(<V::Error as Error>::custom(format!(
-                                        "Invalid key binding, scancode too big: {}",
-                                        scancode
-                                    )));
-                                }
-                                key = Some(Key::Scancode(scancode as u32));
-                            } else {
-                                let k = Key::deserialize(val).map_err(V::Error::custom)?;
-                                key = Some(k);
+                            let value = map.next_value::<SerdeValue>()?;
+                            match value.as_integer() {
+                                Some(scancode) => {
+                                    if scancode > i64::from(u32::MAX) {
+                                        return Err(<V::Error as Error>::custom(format!(
+                                            "Invalid key binding, scancode too big: {}",
+                                            scancode
+                                        )));
+                                    }
+                                    key = Some(Key::Scancode(scancode as u32));
+                                },
```

Comment:
Maybe do explicit check for negative numbers?

Reply:
We need to check for both negative numbers **and** positive numbers that are too big though. But there's no need why we would convert twice, so we can just use `u32::try_from` instead.